pysteps/nowcasts/linear_blending.py

```diff
@@ -0,0 +1,172 @@
+            n_ens_members_nowcast = 1
+
+        if len(R_nwp.shape) == 4:
+            n_ens_members_nwp = R_nwp.shape[0]
+            if n_ens_members_nwp == 1:
+                R_nwp = np.squeeze(R_nwp)
+        else:
+            n_ens_members_nwp = 1
+
+        n_ens_members_max = max(n_ens_members_nowcast, n_ens_members_nwp)
+        n_ens_members_min = min(n_ens_members_nowcast, n_ens_members_nwp)
+
+        if n_ens_members_min != n_ens_members_max:
+            if n_ens_members_nwp == 1:
+                R_nwp = np.repeat(R_nwp[np.newaxis, :, :], n_ens_members_max, axis=0)
+            elif n_ens_members_nowcast == 1:
+                R_nowcast = np.repeat(
+                    R_nowcast[np.newaxis, :, :], n_ens_members_max, axis=0
+                )
+            else:
+                repeats = [
+                    (n_ens_members_max + i) // n_ens_members_min
+                    for i in range(n_ens_members_min)
+                ]
+
+                if n_ens_members_nwp == n_ens_members_min:
+                    R_nwp = np.repeat(R_nwp, repeats, axis=0)
+                elif n_ens_members_nowcast == n_ens_members_min:
+                    R_nowcast = np.repeat(R_nowcast, repeats, axis=0)
+
+        # Check if dimensions are correct
+        assert (
+            R_nwp.shape == R_nowcast.shape
+        ), "The dimensions of R_nowcast and R_nwp need to be identical: dimension of R_nwp = {} and dimension of R_nowcast = {}".format(
+            R_nwp.shape, R_nowcast.shape
+        )
+
+        # Initialise output
+        R_blended = np.zeros_like(R_nowcast)
+
+        # Calculate the weights
+        for i in range(timesteps):
+            # Calculate what time we are at
+            t = (i + 1) * timestep
+
+            # Calculate the weight with a linear relation (weight_nwp at start_blending = 0.0)
+            # and (weight_nwp at end_blending = 1.0)
+            weight_nwp = (t - start_blending) / (end_blending - start_blending)
+
+            # Set weights at times before start_blending and after end_blending
+            if weight_nwp < 0.0:
+                weight_nwp = 0.0
+            elif weight_nwp > 1.0:
+                weight_nwp = 1.0
+
+            # Calculate weight_nowcast
+            weight_nowcast = 1.0 - weight_nwp
+
+            # Calculate output by combining R_nwp and R_nowcast,
+            # while distinguishing between ensemble and non-ensemble methods
+            if n_ens_members_max == 1:
+                R_blended[i, :, :] = (
+                    weight_nwp * R_nwp[i, :, :] + weight_nowcast * R_nowcast[i, :, :]
+                )
+            else:
+                R_blended[:, i, :, :] = (
+                    weight_nwp * R_nwp[:, i, :, :]
+                    + weight_nowcast * R_nowcast[:, i, :, :]
+                )
+
+            # Find where the NaN values are and replace them with NWP data
+            nan_indices = np.isnan(R_blended)
+            R_blended[nan_indices] = R_nwp[nan_indices]
```

Comment:
Is it possible that the NWP data also has NaNs? If so, we can also fill those by the nowcast data.

Reply:
I think this will not occur. Either the NWP data is present or it's not, NaNs are not likely to occur.